Sources/LanguageServerProtocol/SupportTypes/WorkspaceSettings.swift

```diff
@@ -75,3 +80,63 @@ public struct ClangCompileCommand: Codable, Hashable {
     self.workingDirectory = workingDirectory
   }
 }
+
+public struct SourceKitLSPWorkspaceSettings: Decodable, Hashable {
+  private enum CodingKeys: String, CodingKey {
+    case sourcekit_lsp = "sourcekit-lsp"
```

Comment:
Could we lift this custom behaviour into `WorkspaceSettingsChange`? It already needs to have a custom Codable implementation, whereas this struct can just use the auto-generated one once the "sourcekitlsp" namespace is handled.

For example inside `WorkspaceSettingsChange.init(from: Decoder)` you could do

```swift
if let container = decoder.container(keyedBy: CodingKeys.self),
   let settings = container.decode(SourceKitLSPWorkspaceSettings.self, forKey: .sourcekitlsp)
{
  self = .sourcekitlsp(settings)
} else if ...
```